### DISCUSSION TITLE
Same account shows different balance when "without grouping by date" is clicked

### DISCUSSION THREAD

**Discussion by alexklibisz** [firefly-iii/firefly-iii#6602] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have an Asset account that shows an incorrect balance on the default account page (grouped by date) and a different (correct) balance when I click "Show all transactions without grouping them by date." The asset accounts page shows the first, incorrect balance.

Here is the asset accounts page w/ the incorrect balance:

<img width="1281" alt="Screenshot 2022-11-05 at 9 49 43 AM" src="https://user-images.githubusercontent.com/8015228/200125765-fc9951eb-6dac-4a6b-881e-968f3f7123a9.png">

Here is the account page w/ the same incorrect balance:

<img width="1259" alt="Screenshot 2022-11-05 at 9 46 07 AM" src="https://user-images.githubusercontent.com/8015228/200125792-44444378-49ae-4b20-b30b-8e85497b42c6.png">

Here is the account page, expanded, w/ the _correct_ balance:

<img width="1264" alt="Screenshot 2022-11-05 at 9 45 19 AM" src="https://user-images.githubusercontent.com/8015228/200125809-ad49dcc2-8229-4d09-9de7-423da17bfaa2.png">

### Expected behaviour

All places show the same balance

### Debug information

Debug information generated at 2022-11-05 10:48:45 America/New_York for Firefly III version **5.7.15**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.15 |
| Firefly III API | 1.5.6 |
| Build | 298, 02-11-2022 05:00:04 CET |
| DB version | 18 (exp. 18) |
| PHP | `8.1.11` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | America/New_York |
| Browser TZ | America/Chicago |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-11-01 00:00:00 |
| Session end | 2022-11-30 23:59:59 |
| Session first | 2020-01-01 00:00:00 |
| User ID | 3 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 |

### Steps to reproduce

I'm not sure, but see my screenshots for an example

**Comment by alexklibisz** [comment]
Well, it actually shows the correct balance now after rebooting the server and database.

Is there some sort of caching going on? If there is, is there a way to disable it completely?

**Reply by JC5** [reply]
Yes, this is a cache issue. Browsing to `firefly.site/flush` also works to reset it.

**Reply by alexklibisz** [reply]
Is there a way to entirely disable the cache?

When I use firefly, it's usually a steady mix of reads and writes. I don't really care if I wait a few hundred more milliseconds for a page to load. It's much more annoying to see a wrong number and spend time figuring out it's the cache or actually wrong.

**Reply by JC5** [reply]
Try `CACHE_DRIVER=array`

**Reply by alexklibisz** [reply]
Thanks, I set that and will see how it goes.
